### ISSUE TITLE
The list group disapears, leads to "zsh: do you wish to see all X possibilities"

### ISSUE THREAD

**Issue by dmknght** [marlonrichert/zsh-autocomplete#142]
## Environment
* `zsh-autocomplete` f75d296ec218573cb4c14abeebfd952686135b75`
* Zsh version: `debian/5.8-5`
* Framework: "none"
* Plugin manager: "none"
* `~/.zshrc` file: 
```
# Greeting
echo "Welcome to Parrot OS"

# Prompt
PROMPT="%F{red}┌[%f%F{cyan}%m%f%F{red}]─[%f%F{yellow}%D{%H:%M-%d/%m}%f%F{red}]─[%f%F{magenta}%d%f%F{red}]%f"$'\n'"%F{red}└╼%f%F{green}$USER%f%F{yellow}$%f"
# Export PATH$
export PATH=~/.local/bin:/snap/bin:/usr/sandbox/:/usr/local/bin:/usr/bin:/bin:/usr/local/games:/usr/games:/usr/share/games:/usr/local/sbin:/usr/sbin:/sbin:$PATH

# alias
alias ls='ls -lh --color=auto'
alias dir='dir --color=auto'
alias vdir='vdir --color=auto'
alias grep='grep --color=auto'
alias fgrep='fgrep --color=auto'
alias egrep='egrep --color=auto'
#####################################################
# Auto completion / suggestion
# Mixing zsh-autocomplete and zsh-autosuggestions
# Requires: zsh-autocomplete (custom packaging by Parrot Team)
# Jobs: suggest files / foldername / histsory bellow the prompt
# Requires: zsh-autosuggestions (packaging by Debian Team)
# Jobs: Fish-like suggestion for command history
source /usr/share/zsh-autosuggestions/zsh-autosuggestions.zsh
source /usr/share/zsh-autocomplete/zsh-autocomplete.plugin.zsh
# Select all suggestion instead of top on result only
zstyle ':autocomplete:tab:*' insert-unambiguous yes
zstyle ':autocomplete:tab:*' widget-style menu-select
zstyle ':autocomplete:*' max-lines 100%
zstyle ':completion:*' list-rows-first yes
zstyle ':completion:*' group-name ''
zstyle ':autocomplete:*' min-input 2
# Set a bash-like history
bindkey $key[Up] up-line-or-history
bindkey $key[Down] down-line-or-history

##################################################
# Fish like syntax highlighting
# Requires "zsh-syntax-highlighting" from apt
source /usr/share/zsh-syntax-highlighting/zsh-syntax-highlighting.zsh

# Save type history for completion and easier life
HISTFILE=~/.zsh_history
HISTSIZE=10000
SAVEHIST=10000
setopt appendhistory
```

Please test if the bug occurs without config files:
```zsh
$ cd $(mktemp -d)  # Create a temp dir and enter it.
$ ZDOTDIR=$PWD HOME=$PWD zsh -f  # Start a subshell in it without config files.
% source path/to/zsh-autocomplete.plugin.zsh  # Source the plugin.
% # Insert here the steps you take to reproduce your bug.
```

After latest update, with this config file, i get the problem when use `apt show ` like this `zsh: do you wish to see all 46 possibilities (25 lines)?`. It leads to missing character and typing trouble from stdin as i reported before.
I tried disable all `zstyle` and the bug is still there. The list-rows-first seems doesn't work anymore.
# One more random problem
When i use `apt show` and `apt search` command multiple time, there are `zsh` process in backgrounds leads to CPU leak after 30 mins or 1 hours. It is in previous update so i have no result for latest patch. I'll keep testing it and create an other issue if it is still happening

**Comment by dmknght** [comment]
P/s: when i type enter, nothing shows / happens

**Comment by dmknght** [comment]
Update 1: turned out it doesn't work for `apt` completion only and likely it doesn't support `bash completion` in some cases

**Comment by dmknght** [comment]
Update: turned out i upgraded the plugin and old files were kept. Fresh reinstall fixed problem

**Comment by dmknght** [comment]
@marlonrichert well it is happening again, after i upgraded latest commit yesterday. Fresh install of plugin done so i think i should repopen this issue.
Anyway i tested with 
```
zstyle ':completion:*' list-rows-first yes
zstyle ':completion:*' group-name ''
```
and nothing changed. In other cases, the group list works just fine.
![image](https://user-images.githubusercontent.com/29118926/101524836-5afd1800-3982-11eb-8458-4bbbd6df04e7.png)

**Comment by marlonrichert** [comment]
@dmknght I still cannot reproduce this, but as I said before, I don't have access to `apt`.

When you get that question, what output do you get when you press `y`?

**Comment by marlonrichert** [comment]
I tried to simulate it as follows:
```zsh
% apt() {}
% apt-cache() { print -l loremipsumfoobarbaz{1..1000} }
% apt show # then press Tab
```
Works fine for me, but perhaps this is not realistic data.

Could you please run `apt-cache --generate pkgnames` and paste the results here? Then I can test with it.

**Comment by dmknght** [comment]
> I tried to mock test it as follows:
> 
> ```shell
> % apt() {}
> % apt-cache() { print -l loremipsumfoobarbaz{1..1000} }
> % apt show # then press Tab
> ```
> 
> Works fine for me, but perhaps this is not realistic data.
> 
> Could you please run `apt-cache --generate pkgnames` and paste the results here? Then I can test with it.

the command you gave showed error for me but i think you can use the index file of `apt` for the test?
URL: `https://deb.parrot.sh/parrot/dists/rolling/InRelease`.
Could you try install debian based distro in VM and test with plugin?

**Comment by marlonrichert** [comment]
You haven't answered me yet: When you get the question, what output do you get from pressing `y`?

Also, if you press `n` and you then press `^Xh`, what output do you get?

> the command you gave showed error for me

What kind of error? Can you paste it here?

> i think you can use the index file of apt for the test?
URL: https://deb.parrot.sh/parrot/dists/rolling/InRelease.

That doesn't contain any package names.

> Could you try install debian based distro in VM and test with plugin?

Could, but that's quite a lot of effort, since I don't normally use VMs for anything. I wouldn't know where to start. 🙂

**Comment by dmknght** [comment]
> You haven't answered me yet: When you get the question, what output do you get from pressing `y`?
> 
> Also, if you press `n` and you then press `^Xh`, what output do you get?
> 
> > the command you gave showed error for me
> 
> What kind of error? Can you paste it here?
> 
> > i think you can use the index file of apt for the test?
> > URL: https://deb.parrot.sh/parrot/dists/rolling/InRelease.
> 
> That doesn't contain any package names.
> 
> > Could you try install debian based distro in VM and test with plugin?
> 
> Could, but that's quite a lot of effort, since I don't normally use VMs for anything. I wouldn't know where to start. slightly_smiling_face

Sorry my bad. This url should contains Package name `https://deb.parrot.sh/parrot/dists/rolling/main/binary-amd64/Packages`.
if i get the question, press `y` or enter doesn't show anything as i remember. but well somehow i can't reprocedure it on my system after reboot. It is weird.

**Comment by marlonrichert** [comment]
If after typing `apt show r`, you press `^Xh`, what output do you get?

**Comment by dmknght** [comment]
> If after typing `apt show r`, you press `^Xh`, what output do you get?

unfortunately this time i get nothing from "do you wish to see ..."
But there is an other problem: There are so many zsh processes in background
![image](https://user-images.githubusercontent.com/29118926/101676433-142b2300-3a53-11eb-85d7-9a6068e001c0.png)

Let me try clean my installation of the plugin again to make sure there is nothing wrong with my installation.

**Comment by marlonrichert** [comment]
After typing `apt show r`, can you please press `^Xh` and show me what output you get? Doesn't matter if there's no question shown.

**Comment by dmknght** [comment]
> After typing `apt show r`, can you please press `^Xh` and show me what output you get? Doesn't matter if there's no question shown.

Oh sorry i misunderstood it. Here is the screenshot. It was `Ctrl + X` then `h` am i right?
![image](https://user-images.githubusercontent.com/29118926/101676935-c6fb8100-3a53-11eb-990f-cc3994520d2e.png)

**Comment by marlonrichert** [comment]
OK, can you now do the following?
```zsh
% functions -T _deb_packages _apt-cmd_sm_sm _apt-cmd_sm _apt-cmd _apt _deb_packages_update_avail _cache_invalid
% apt show r # press Ctrl-Space
```

**Comment by dmknght** [comment]
> OK, can you now do the following?
> 
> ```shell
> % functions -T _deb_packages _apt-cmd_sm_sm _apt-cmd_sm _apt-cmd _apt _deb_packages_update_avail _cache_invalid
> % apt show r # press Ctrl-Space
> ```

This is really long. The full output:
```
┌[ParrotPC]─[02:24-10/12]─[/home/dmknght]
└╼dmknght$apt show r+_apt:1> case apt (apt)
+_apt:2> _apt-cmdess Ctrl+Space to expand)
+_apt-cmd:1> local expl_action expl_packages subcmd
+_apt-cmd:2> _description actions expl_action action      
+_apt-cmd:3> _description packages expl_packages packageUp for more)
+_apt-cmd:5> _apt-cmd_smpackages _apt-cmd_sm_sm _apt-cmd_sm _apt-cmd _apt _deb_
+_apt-cmd_sm:1> typeset -A canonicalize num_options
+_apt-cmd_sm:2> canonicalize=( --help -h,--help -t -t,--target-release -c -c,--config-file --target-release -t,--target-release --option -o,--option -v -v,--version -h -h,--help -o -o,--option --version -v,--version --config-file -c,--config-file ) 
+_apt-cmd_sm:3> num_options=( -c,--config-file 1 -v,--version 1 -o,--option -1 -h,--help 1 -t,--target-release 1 ) 
+_apt-cmd_sm:5> local short_hasarg short_bool short_intlevel short_configfile short_arbitem
+_apt-cmd_sm:6> local long_hasarg long_bool long_intlevel long_configfile long_arbitem
+_apt-cmd_sm:7> local short_release long_release
+_apt-cmd_sm:8> local bool_prefix
+_apt-cmd_sm:9> short_hasarg=( ) 
+_apt-cmd_sm:10> short_bool=( h v ) 
+_apt-cmd_sm:11> short_intlevel=( ) 
+_apt-cmd_sm:12> short_configfile=( c ) 
+_apt-cmd_sm:13> short_arbitem=( o ) 
+_apt-cmd_sm:14> short_release=( t ) 
+_apt-cmd_sm:15> long_hasarg=( ) 
+_apt-cmd_sm:16> long_bool=( help version ) 
+_apt-cmd_sm:17> long_intlevel=( ) 
+_apt-cmd_sm:18> long_configfile=( config-file ) 
+_apt-cmd_sm:19> long_arbitem=( option ) 
+_apt-cmd_sm:20> long_release=( target-release ) 
+_apt-cmd_sm:21> bool_prefix=( --no- --false- --without- --off- --disable- --yes- --true- --with- --on- --enable- ) 
+_apt-cmd_sm:23> local current_option tmp1 tmp2 tmp3
+_apt-cmd_sm:25> _apt-cmd_sm_sm
+_apt-cmd_sm_sm:1> local _ra_p1 _ra_p2 _ra_left _ra_right _ra_com expl tmp nm=0
+_apt-cmd_sm_sm:2> local _ra_actions _ra_line=$'apt\C-@show\C-@r'
+_apt-cmd_sm_sm:3> _ra_actions=( ) 
+_apt-cmd_sm_sm:4> zregexparse -c _ra_p1 _ra_p2 $'apt\C-@show\C-@r' $'/[^\C-@]#\C-@/' '(' /--/+ '(' '/(help|version)=/' '-_apt_consume_long ${match[1]%=}' '(' $'/[^\C-@]##\C-@/' ':_ra_comp $'\''names:boolean value:(no false without off disable yes true with on enable)'\' '|' $'/\C-@/' $'/[^\C-@]#\C-@/' ':_ra_comp $'\''names:boolean value:(no false without off disable yes true with on enable)'\' ')' '|' $'/(help|version)\C-@/' '-_apt_consume_long ${match[1]%$'\''\0'\''}' $'/((no|false|without|off|disable|yes|true|with|on|enable)\C-@|)/' ':_ra_comp $'\''names:boolean value:(no false without off disable yes true with on enable)'\' '|' $'/(config-file)\C-@/' '-_apt_consume_long ${match[1]%$'\''\0'\''}' $'/[^\C-@]#\C-@/' ':_ra_comp $'\''files:config file:_files'\' '|' '/(config-file)=/' '-_apt_consume_long ${match[1]%=}' '(' $'/[^\C-@]##\C-@/' ':_ra_comp $'\''files:config file:_files'\' '|' $'/\C-@/' $'/[^\C-@]#\C-@/' ':_ra_comp $'\''files:config file:_files'\' ')' '|' $'/(option)\C-@/' '-_apt_consume_long ${match[1]%$'\''\0'\''}' $'/[^\C-@]#\C-@/' ':_ra_comp $'\''options:option:'\' '|' '/(option)=/' '-_apt_consume_long ${match[1]%=}' '(' $'/[^\C-@]##\C-@/' ':_ra_comp $'\''options:option:'\' '|' $'/\C-@/' $'/[^\C-@]#\C-@/' ':_ra_comp $'\''options:option:'\' ')' '|' $'/(target-release)\C-@/' '-_apt_consume_long ${match[1]%$'\''\0'\''}' $'/[^\C-@]#\C-@/' ':_ra_comp $'\''releases:release name:_apt_releases'\' '|' '/(target-release)=/' '-_apt_consume_long ${match[1]%=}' '(' $'/[^\C-@]##\C-@/' ':_ra_comp $'\''releases:release name:_apt_releases'\' '|' $'/\C-@/' $'/[^\C-@]#\C-@/' ':_ra_comp $'\''releases:release name:_apt_releases'\' ')' '|' '/(no|false|without|off|disable|yes|true|with|on|enable)-/+' '(' '/(h|v)=/' '-_apt_consume_short ${match[1][-2]}' '(' $'/[^\C-@]##\C-@/' ':_ra_comp $'\''names:boolean value:(no false without off disable yes true with on enable)'\' '|' $'/\C-@/' $'/[^\C-@]#\C-@/' ':_ra_comp $'\''names:boolean value:(no false without off disable yes true with on enable)'\' ')' '|' $'/(h|v)\C-@/' '-_apt_consume_short ${match[1][-2]}' $'/((no|false|without|off|disable|yes|true|with|on|enable)\C-@|)/' ':_ra_comp $'\''names:boolean value:(no false without off disable yes true with on enable)'\' '|' $'/(c)\C-@/' '-_apt_consume_short ${match[1][-2]}' $'/[^\C-@]#\C-@/' ':_ra_comp $'\''files:config file:_files'\' '|' '/(c)=/' '-_apt_consume_short ${match[1][-2]}' '(' $'/[^\C-@]##\C-@/' ':_ra_comp $'\''files:config file:_files'\' '|' $'/\C-@/' $'/[^\C-@]#\C-@/' ':_ra_comp $'\''files:config file:_files'\' ')' '|' $'/(o)\C-@/' '-_apt_consume_short ${match[1][-2]}' $'/[^\C-@]#\C-@/' ':_ra_comp $'\''options:option:'\' '|' '/(o)=/' '-_apt_consume_short ${match[1][-2]}' '(' $'/[^\C-@]##\C-@/' ':_ra_comp $'\''options:option:'\' '|' $'/\C-@/' $'/[^\C-@]#\C-@/' ':_ra_comp $'\''options:option:'\' ')' '|' $'/(t)\C-@/' '-_apt_consume_short ${match[1][-2]}' $'/[^\C-@]#\C-@/' ':_ra_comp $'\''releases:release name:_apt_releases'\' '|' '/(t)=/' '-_apt_consume_short ${match[1][-2]}' '(' $'/[^\C-@]##\C-@/' ':_ra_comp $'\''releases:release name:_apt_releases'\' '|' $'/\C-@/' $'/[^\C-@]#\C-@/' ':_ra_comp $'\''releases:release name:_apt_releases'\' ')' '|' '/(help|version)=/' '-_apt_consume_long ${match[1]%=}' '(' $'/[^\C-@]##\C-@/' ':_ra_comp $'\''names:boolean value:(no false without off disable yes true with on enable)'\' '|' $'/\C-@/' $'/[^\C-@]#\C-@/' ':_ra_comp $'\''names:boolean value:(no false without off disable yes true with on enable)'\' ')' '|' $'/(help|version)\C-@/' '-_apt_consume_long ${match[1]%$'\''\0'\''}' $'/((no|false|without|off|disable|yes|true|with|on|enable)\C-@|)/' ':_ra_comp $'\''names:boolean value:(no false without off disable yes true with on enable)'\' '|' $'/(config-file)\C-@/' '-_apt_consume_long ${match[1]%$'\''\0'\''}' $'/[^\C-@]#\C-@/' ':_ra_comp $'\''files:config file:_files'\' '|' '/(config-file)=/' '-_apt_consume_long ${match[1]%=}' '(' $'/[^\C-@]##\C-@/' ':_ra_comp $'\''files:config file:_files'\' '|' $'/\C-@/' $'/[^\C-@]#\C-@/' ':_ra_comp $'\''files:config file:_files'\' ')' '|' $'/(option)\C-@/' '-_apt_consume_long ${match[1]%$'\''\0'\''}' $'/[^\C-@]#\C-@/' ':_ra_comp $'\''options:option:'\' '|' '/(option)=/' '-_apt_consume_long ${match[1]%=}' '(' $'/[^\C-@]##\C-@/' ':_ra_comp $'\''options:option:'\' '|' $'/\C-@/' $'/[^\C-@]#\C-@/' ':_ra_comp $'\''options:option:'\' ')' '|' $'/(target-release)\C-@/' '-_apt_consume_long ${match[1]%$'\''\0'\''}' $'/[^\C-@]#\C-@/' ':_ra_comp $'\''releases:release name:_apt_releases'\' '|' '/(target-release)=/' '-_apt_consume_long ${match[1]%=}' '(' $'/[^\C-@]##\C-@/' ':_ra_comp $'\''releases:release name:_apt_releases'\' '|' $'/\C-@/' $'/[^\C-@]#\C-@/' ':_ra_comp $'\''releases:release name:_apt_releases'\' ')' '|' '/[]/' ':_ra_comp $'\''options:option:{tmp1="${(j:|:)${(@)${(@M)${(@s:,:)${(@kj:,:)num_options[(R)*~0]}}:#--*}#--}}"\ntmp2=($_ra_left${(M)^long_bool:#$~tmp1} $_ra_left${(M)^long_intlevel:#$~tmp1})\ntmp3=($_ra_left${(M)^long_hasarg:#$~tmp1} $_ra_left${(M)^long_configfile:#$~tmp1} $_ra_left${(M)^long_arbitem:#$~tmp1} $_ra_left${(M)^long_release:#$~tmp1})\ntmp1="${(j:|:)${(@)${(@M)${(@s:,:)${(@kj:,:)num_options[(R)*~0]}}:#-?}#-}}"\ntmp2=("$tmp2[@]" $_ra_left${(M)^short_bool:#$~tmp1} $_ra_left${(M)^short_intlevel:#$~tmp1})\ntmp3=("$tmp3[@]" $_ra_left${(M)^short_hasarg:#$~tmp1} $_ra_left${(M)^short_configfile:#$~tmp1} $_ra_left${(M)^short_arbitem:#$~tmp1} $_ra_left${(M)^short_release:#$~tmp1})\n_describe -o option tmp2 -- tmp3 -S=}'\' ')' ')' '|' /-/+ '(' $'/(h|v|)#(h|v)(\C-@(no|false|without|off|disable|yes|true|with|on|enable)|(no|false|without|off|disable|yes|true|with|on|enable)|)\C-@/' '-_apt_consume_short ${match[1]%%($'\''\0'\''(no|false|without|off|disable|yes|true|with|on|enable)|(no|false|without|off|disable|yes|true|with|on|enable)|)$'\''\0'\''}' '|' '/(h|v|)#(h|v)=/' '-_apt_consume_short ${match[1]%=}' '(' $'/[^\C-@]##\C-@/' ':_ra_comp $'\''names:boolean value:(no false without off disable yes true with on enable)'\' '|' $'/\C-@/' $'/[^\C-@]#\C-@/' ':_ra_comp $'\''names:boolean value:(no false without off disable yes true with on enable)'\' ')' '|' '/(h|v|)#(c)(=|)/' '-_apt_consume_short ${match[1]%=}' '(' $'/[^\C-@]##\C-@/' ':_ra_comp $'\''files:config file:_files'\' '|' $'/\C-@/' $'/[^\C-@]#\C-@/' ':_ra_comp $'\''files:config file:_files'\' ')' '|' '/(h|v|)#(o)(=|)/' '-_apt_consume_short ${match[1]%=}' '(' $'/[^\C-@]##\C-@/' ':_ra_comp $'\''options:option:'\' '|' $'/\C-@/' $'/[^\C-@]#\C-@/' ':_ra_comp $'\''options:option:'\' ')' '|' '/(h|v|)#(t)(=|)/' '-_apt_consume_short ${match[1]%=}' '(' $'/[^\C-@]##\C-@/' ':_ra_comp $'\''releases:release name:_apt_releases'\' '|' $'/\C-@/' $'/[^\C-@]#\C-@/' ':_ra_comp $'\''release name:release:_apt_releases'\' ')' '|' '/[]/' ')' '|' '/[]/' ':_ra_comp $'\''options:option:{{ ! zstyle -T ":completion:${curcontext}:options" prefix-needed || [[ "$PREFIX" = -* ]] } && { {if [[ $PREFIX = -(h|v|)# ]]; then\n  _apt_consume_short ${PREFIX[2,-1]}\n  tmp1=(${(M)${(s:,:)${(kj:,:)num_options[(R)*~0]}}:#-?})\n  tmp2=(${PREFIX}${^tmp1#-})\n  _describe -o option tmp1 tmp2\nelif [[ -z "$PREFIX" ]]; then\n  tmp1=(${(M)${(s:,:)${(kj:,:)num_options[(R)*~0]}}:#-?})\n  _describe -o option tmp1\nfi}; tmp1="${(j:|:)${(@)${(@M)${(@s:,:)${(@kj:,:)num_options[(R)*~0]}}:#--*}#--}}"\ntmp2=(--${(M)^long_bool:#$~tmp1} --${(M)^long_intlevel:#$~tmp1})\ntmp3=(--${(M)^long_hasarg:#$~tmp1} --${(M)^long_configfile:#$~tmp1} --${(M)^long_arbitem:#$~tmp1} --${(M)^long_release:#$~tmp1})\n_describe -o option tmp2 -- tmp3 -S= -- bool_prefix -S "" }}'\' ')' '#' '(' $'/list\C-@/' '(' '(' $'/--installed\C-@/' ':_ra_comp $'\''packages::_deb_packages "$expl_packages[@]" installed'\' '#' '/[]/' '|' // %-% $'/--[^\C-@]#\C-@/' ':_ra_comp $'\''options:option:(--installed --upgradable --all-versions)'\' '|' ')' $'/[^-\C-@][^\C-@]#\C-@/' ':_ra_comp $'\''packages::_deb_packages "$expl_packages[@]" avail'\' '#' ')' '|' $'/search\C-@/' $'/[^\C-@]#\C-@/' ':_ra_comp $'\''strings:pattern:'\' '|' $'/showsrc\C-@/' $'/[^\C-@]#\C-@/' ':_ra_comp $'\''packages::_deb_packages "$expl_packages[@]" avail'\' '#' '|' $'/show\C-@/' $'/[^\C-@]#\C-@/' ':_ra_comp $'\''packages::_deb_packages "$expl_packages[@]" avail'\' '#' '|' $'/depends\C-@/' $'/[^\C-@]#\C-@/' ':_ra_comp $'\''packages::_deb_packages "$expl_packages[@]" avail'\' '#' '|' $'/rdepends\C-@/' $'/[^\C-@]#\C-@/' ':_ra_comp $'\''packages::_deb_packages "$expl_packages[@]" avail'\' '#' '|' $'/policy\C-@/' $'/[^\C-@]#\C-@/' ':_ra_comp $'\''packages::_deb_packages "$expl_packages[@]" avail'\' '#' '|' $'/update\C-@/' '|' '(' $'/(install|download|source|build-dep)\C-@/' '-subcmd=${match%?}' $'/[^\C-@]#\C-@/' ':_ra_comp $'\''packages::_deb_packages -qr "\\n\\t\\- /=+-" "$expl_packages[@]" avail'\' '#' '(' // '%(.|)/%' '-[[ $subcmd = install ]]' '/[]/' ':_ra_comp $'\''files: :_deb_files -c'\' '|' $'/[^\C-@/=]#//' '/[]/' ':_ra_comp $'\''apt-releases:release name:_apt_releases'\' '|' $'/[^\C-@/=]#=/' '/[]/' ':_ra_comp $'\''apt-package-versions:package version:_apt_versions_of_binary_package'\' '|' ')' ')' '|' $'/(remove|reinstall|purge)\C-@/' $'/[^\C-@]#\C-@/' ':_ra_comp $'\''packages::_deb_packages "$expl_packages[@]" installed'\' '#' '|' $'/upgrade\C-@/' '|' $'/autoclean\C-@/' '|' $'/changelog\C-@/' $'/[^\C-@]#\C-@/' ':_ra_comp $'\''packages::_deb_packages "$expl_packages[@]" avail'\' '#' '|' $'/autoremove\C-@/' '|' $'/full-upgrade\C-@/' '|' $'/dist-upgrade\C-@/' '|' $'/edit-sources\C-@/' '|' '/[]/' ':_ra_comp $'\''argument-1::compadd "$expl_action[@]" list search showsrc show depends rdepends policy update install reinstall download source build-dep remove upgrade full-upgrade dist-upgrade edit-sources autoclean changelog autoremove purge'\' ')'
+_apt-cmd_sm_sm:4> _ra_comp 'packages::_deb_packages "$expl_packages[@]" avail'
+_apt-cmd_sm_sm:5> case 1 (0 | 2)
+_apt-cmd_sm_sm:5> case 1 (1)
+_apt-cmd_sm_sm:8> [[ r = ** ]]
+_apt-cmd_sm_sm:11> _ra_left='' 
+_apt-cmd_sm_sm:12> _ra_right=r 
+_apt-cmd_sm_sm:13> compset -p 0
+_apt-cmd_sm_sm:14> ((  1  ))
+_apt-cmd_sm_sm:14> _alternative 'packages::_deb_packages "$expl_packages[@]" avail'
+_deb_packages:1> local command=avail expl cachevar pkgset update_policy
+_deb_packages:3> zstyle -s ':completion:*:*:apt:*' cache-policy update_policy
+_deb_packages:4> [[ -z _debs_caching_policy ]]
+_deb_packages:8> [[ avail = (installed|deinstalled|xinstalled|held|uninstalled|avail|available|source) ]]
+_deb_packages:13> zstyle -s :completion:list-expand:complete:apt:: packageset pkgset
+_deb_packages:15> [[ '' = (installed|deinstalled|xinstalled|held|uninstalled|avail|available|source) ]]
+_deb_packages:16> pkgset=avail 
+_deb_packages:19> [[ avail = available ]]
+_deb_packages:21> expl=( -M 'r:|?=**' -M 'l:|=** m:{[:upper:][:lower:]-_}={[:lower:][:upper:]_-}' -o nosort -J packages -M 'r:|?=**' -M 'l:|=** m:{[:upper:][:lower:]-_}={[:lower:][:upper:]_-}' -o nosort -J packages -X $'%{\C-[[01;02;39m%}package%{\C-[[0m%}' ) 
+_deb_packages:23> _deb_packages_update_avail
+_deb_packages_update_avail:1> [[ 1 -eq 0 ]]
+_deb_packages_update_avail:2> _cache_invalid DEBS_avail
+_cache_invalid:5> local _cache_ident _cache_dir _cache_path _cache_policy
+_cache_invalid:6> _cache_ident=DEBS_avail 
+_cache_invalid:10> zstyle -t :completion:list-expand:complete:apt:: use-cache
+_cache_invalid:12> zstyle -s :completion:list-expand:complete:apt:: cache-path _cache_dir
+_cache_invalid:13> : /home/dmknght/.cache/zcompcache
+_cache_invalid:14> _cache_path=/home/dmknght/.cache/zcompcache/DEBS_avail 
+_cache_invalid:18> zstyle -s :completion:list-expand:complete:apt:: cache-policy _cache_policy
+_cache_invalid:19> [[ -n _debs_caching_policy ]]
+_cache_invalid:19> _debs_caching_policy /home/dmknght/.cache/zcompcache/DEBS_avail
+_cache_invalid:21> return 1
+_deb_packages_update_avail:10> cachevar=_deb_packages_cache_avail 
+_deb_packages:25> typeset -gH _deb_packages_cache_avail
+_deb_packages:27> _tags packages
+_deb_packages:27> compadd -M 'r:|?=**' -M 'l:|=** m:{[:upper:][:lower:]-_}={[:lower:][:upper:]_-}' -o nosort -J packages -M 'r:|?=**' -M 'l:|=** m:{[:upper:][:lower:]-_}={[:lower:][:upper:]_-}' -o nosort -J packages -X $'%{\C-[[01;02;39m%}package%{\C-[[0m%}' -a - _deb_packages_cache_avail
```
Then after waiting (while i was typing this), it went to `apt show libgkarrays` automatically

**Comment by marlonrichert** [comment]
Can you list the contents of `/home/dmknght/.cache/zcompcache/DEBS_avail`?

**Comment by dmknght** [comment]
> Can you list the contents of `/home/dmknght/.cache/zcompcache/DEBS_avail`?

well it is really long. Let me zip it and send to you
p/s quick question: the file zcompdump something moved to the folder `.cache/zcompcache/`?

**Comment by dmknght** [comment]
[DEBS_avail.zip](https://github.com/marlonrichert/zsh-autocomplete/files/5668198/DEBS_avail.zip)
Here is the file

**Comment by marlonrichert** [comment]
> p/s quick question: the file zcompdump something moved to the folder `.cache/zcompcache/`?

Yes, I moved it from `$ZDOTDIR` to `$XDG_CACHE_HOME`.

**Comment by dmknght** [comment]
> Yes, I moved it from `$ZDOTDIR` to `$XDG_CACHE_HOME`.

nice! So i should notice this folder more when i have any problem.
Anyway do you need any information to debug? I'm ready to do anything that can help you :D

**Comment by marlonrichert** [comment]
I just have to try some things now with the data you provided. I'll let you know when I need anything. Thanks!

**Comment by marlonrichert** [comment]
The problem appears to be that there are _70124_ packages in the cache, of which _37719_ have the letter `r` in them. Just after all of these have been processed by the `compadd` builtin, Zsh freezes completely. I presume because it's busy freeing up a bunch of memory.

I'll see if I can work around it somehow.

**Comment by dmknght** [comment]
> The problem appears to be that there are **_70124_** packages in the cache, of which **_37719_** have the letter `r` in them. Just after all of these have been processed by the `compadd` builtin (but not immediately), Zsh freezes completely. I presume because it's busy (de)allocating up a bunch of memory.
> 
> I'll see if I can work around it somehow.

I think, in this case, users may want package that starts with `r` only and same behavior with other packages. But ofc, suggestion for other name like `rad` for `radare2` and `xxradxxx` will be better than only starts with. I think maybe you can add an option that suggest strring that starts with user's keystroke only? But for how zsh keeps checking text and hanging, it must have other way.

**Comment by dmknght** [comment]
@marlonrichert something about the "X possibilities"' has just happened
![image](https://user-images.githubusercontent.com/29118926/101818944-7ef46100-3b1c-11eb-8bcc-60896c797423.png)
When i use Ctrl + x then h, ofc it showed `No pattern completions found.` The problem is it makes the typo have the missing character problem

**Comment by marlonrichert** [comment]
> I think, in this case, users may want package that starts with `r` only and same behavior with other packages. But ofc, suggestion for other name like `rad` for `radare2` and `xxradxxx` will be better than only starts with. I think maybe you can add an option that suggest strring that starts with user's keystroke only? But for how zsh keeps checking text and hanging, it must have other solution.

That would help only a tiny bit. The real problem is that Zsh's completion system simply doesn't scale when it has to process thousands of matches at once. However, I've found a workaround. I'll push in a fix soon.

**Comment by marlonrichert** [comment]
All right, I tested it and it's all fixed now. Enjoy!

**Comment by dmknght** [comment]
@marlonrichert confirmed it was fixed for `apt`. However the `grep` is having this problem
![image](https://user-images.githubusercontent.com/29118926/101865882-5ea1c200-3b6f-11eb-9af0-cc6ef8937bae.png)
Using ctrl + x -> h
![image](https://user-images.githubusercontent.com/29118926/101865909-6bbeb100-3b6f-11eb-9bf9-321a8fbe4caf.png)
If i press Y
![image](https://user-images.githubusercontent.com/29118926/101865961-901a8d80-3b6f-11eb-969f-cb47a4d42ea3.png)

Fully output
```
┌[ParrotPC]─[12:12-11/12]─[/home/dmknght]
└╼dmknght$grep -r "Applications"
(partial list; press Ctrl+Space to expand)
pattern
history word
>
|
0trace
AAAAB3NzaC1yc2EAAAADAQABAAACAQCloWyJCwct0+urOQJ+ECsuIbctLNlMk6BLDCgZNn7iJzGG5Gi8sCzPfTpYk/3gs42n0FyJhsjPjRVUFAt358X9wAct8ewCDp6yFkdJNMEHnAGtTTD8q97Ze/HY5X+Dwh8HvpvWU/ljPmLsWTie6cfScna2qbFzb5JnnKkheXf2CIIYBbmSjJ3fvfW39cewjxUwVtBsBDEKnP0s39d
amap
Applications
DBManager
dmknght@parrotsec.org
Exploit Search
gdb
KLAV_DB_Manager
-l
manager
-r
remotes/origin/
remotes/origin/master
signature
/tmp/todo
TODO
upstream
uscan
wc
```
Can you reopen this issue please because it is happening for this command. And i think maybe it could happen for other commands like `find`.

**Comment by marlonrichert** [comment]
@dmknght Actually, can you just open a new issue for those? Because it's an entirely different problem.

**Comment by dmknght** [comment]
> @dmknght Actually, can you please open a new issue for those? Because it's an entirely different problem.

Oh sorry i didn't check this comment. I'm testing it with latest version of plugin and it i got any issue i'll create new issue.
